tests/phpunit/tests/style-engine/wpStyleEngineCssRulesStore.php

```diff
@@ -0,0 +1,170 @@
+<?php
+/**
+ * Tests the Style Engine CSS Rules Store class.
+ *
+ * @package WordPress
+ * @subpackage StyleEngine
+ * @since 6.1.0
+ *
+ * @group style-engine
+ */
+
+/**
+ * Tests for registering, storing and retrieving a collection of CSS Rules (a store).
+ *
+ * @coversDefaultClass WP_Style_Engine_CSS_Rules_Store
+ */
+class Tests_Style_Engine_wpStyleEngineCSSRulesStore extends WP_UnitTestCase {
+	/**
+	 * Tear down after each test.
+	 */
+	public function tear_down() {
+		WP_Style_Engine_CSS_Rules_Store::remove_all_stores();
+		parent::tear_down();
+	}
+
+	/**
+	 * Tests creating a new store on instantiation.
+	 *
+	 * @ticket 56467
+	 * @covers ::__construct
+	 */
+	public function test_should_create_new_store_on_instantiation() {
+		$new_pancakes_store = WP_Style_Engine_CSS_Rules_Store::get_store( 'pancakes-with-strawberries' );
+		$this->assertInstanceOf( 'WP_Style_Engine_CSS_Rules_Store', $new_pancakes_store );
+	}
+
+	/**
+	 * Tests that a `$store_name` argument is required and no store will be created without one.
+	 *
+	 * @ticket 56467
+	 * @covers ::get_store
+	 */
+	public function test_should_not_create_store_without_a_store_name() {
+		$not_a_store = WP_Style_Engine_CSS_Rules_Store::get_store( '' );
+		$this->assertEmpty( $not_a_store, 'get_store() did not return an empty value with empty string as argument.' );
+
+		$also_not_a_store = WP_Style_Engine_CSS_Rules_Store::get_store( 123 );
+		$this->assertEmpty( $also_not_a_store, 'get_store() did not return an empty value with number as argument.' );
+
+		$definitely_not_a_store = WP_Style_Engine_CSS_Rules_Store::get_store( null );
+		$this->assertEmpty( $definitely_not_a_store, 'get_store() did not return an empty value with `null` as argument.' );
+	}
+
+	/**
+	 * Tests returning a previously created store when the same selector key is passed.
+	 *
+	 * @ticket 56467
+	 * @covers ::get_store
+	 */
+	public function test_should_return_existing_store() {
+		$new_fish_store = WP_Style_Engine_CSS_Rules_Store::get_store( 'fish-n-chips' );
+		$selector       = '.haddock';
+
+		$new_fish_store->add_rule( $selector );
+		$this->assertEquals( $selector, $new_fish_store->add_rule( $selector )->get_selector(), 'Selector string of store rule does not match expected value' );
```

Comment:
`assertEquals()` should only be used for objects or arrays containing objects. Otherwise, a more appropriate assertion, or `assertSame()` should be used. Applies to all instances in the PR.

- Try replacing all `assertEquals()` with `assertSame()`, then checking any failures to see if they are `object|object[]`.
- Always use the strictest possible assertion.